service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -210,7 +217,8 @@ public ResponseEntity<ApiGcpGcsBucketResource> createBucketReference(
     // Construct a ReferenceGcsBucketResource object from the API input
     var resource =
         ReferencedGcsBucketResource.builder()
-            .wsmResourceFields(getWsmResourceFields(workspaceUuid, body.getMetadata()))
+            .wsmResourceFields(
+                getWsmResourceFields(workspaceUuid, body.getMetadata(), getUserEmail(userRequest)))
```

Comment:
It might be better to pass the `userRequest` in instead of the email. Let that code do the mapping (if needed) rather than have it at each call point?

Reply:
hmm i can't inject SamService into wsmResourceFields though